foundation-extensions/mobile-core/mobile-core-api-reference.md

```diff
@@ -1250,566 +1873,524 @@ public unsafe static void SetLargeIconResourceID (int resourceID);
 
 **iOS syntax**
 
 ```csharp
-ACPCore.Log(ACPMobileLogLevel.Error, "xamarin tag", "xamarin message");
+public static void TrackState (string state, NSMutableDictionary<NSString, NSString> data);
 ```
 
-```text
-[AdobeExperienceSDK ERROR <xamarin tag>]: xamarin message
-```
+* _state_ contains the name of the state to track.
+* _contextData_ contains the context data to attach on the hit.
 
 **Android syntax**
 
 ```csharp
-ACPCore.Log(LoggingMode.Error, "xamarin tag", "xamarin message");
-```
-
-```text
-[AdobeExperienceSDK] xamarin tag - xamarin message
+public unsafe static void TrackState (string state, IDictionary<string, string> contextData);
 ```
-{% endtab %}
-{% endtabs %}
-
-## Handle open URL action
-
-Mobile SDK allows you to add a callback function that is triggered before the [`open url`](https://aep-sdks.gitbook.io/docs/using-mobile-extensions/mobile-core/rules-engine#consequence-types) action occurs. If the callback function returns **Yes**, the SDK does not complete the `open url` action. If the callback function returns **No**, the SDK completes the `open url` action.
-
-{% tabs %}
-{% tab title="iOS \(ACP 2.x\)" %}
-**Objective C**
 
-### registerURLHandler
+* _state_ contains the name of the state to track.
+* _contextData_ contains the context data to attach on the hit.
 
-**Syntax**
+**iOS example**
 
-```objectivec
-+ (void) registerURLHandler: (nonnull BOOL (^) (NSString* __nullable url)) callback;
+```csharp
+var data = new NSMutableDictionary<NSString, NSString>
+{
+  ["key"] = new NSString("value")
+};
+ACPCore.TrackState("state", data);
 ```
 
-**Example**
+**Android example**
 
-```objectivec
-[ACPCore registerURLHandler:^BOOL(NSString * _Nullable url) {
-    ...
-}];
+```csharp
+var data = new Dictionary<string, string>();
+data.Add("key", "value");
+ACPCore.TrackState("state", data);
 ```
 {% endtab %}
 {% endtabs %}
 
-## Set app group \(iOS only\)
 
-You can use the `setAppGroup` method to set the app group, which is used to share user defaults and files among the containing app and the extension apps.
 
-{% hint style="info" %}
-This API _must_ be called in `AppDidFinishLaunching` and before any other interactions with the Adobe Experience SDK have happened. Only the first call to this function will have any effect.
-{% endhint %}
 
-{% tabs %}
-{% tab title="iOS \(ACP 2.x\)" %}
-**Objective-C**
 
-### setAppGroup
 
-**Syntax**
 
-```swift
-public static func setAppGroup(_ group: String?)
-```
 
-**Example**
 
-```text
-[ACPCore setAppGroup:@"app-group-id"];
-```
 
-**Swift**
 
-### setAppGroup
 
-**Syntax**
 
-```swift
-public static func setAppGroup(_ group: String?)
-```
 
-**Example**
 
-```swift
-ACPCore.setAppGroup("app-group-id")
-```
-{% endtab %}
 
-{% tab title="Xamarin" %}
-**C\#**
 
-### setAppGroup
 
-**Syntax**
 
-```csharp
-public static void SetAppGroup (string appGroup);
-```
 
-**Example**
 
-```csharp
-ACPCore.SetAppGroup("app_group");
-```
-{% endtab %}
-{% endtabs %}
+
 
 ## Public classes
```

Comment:
looks like this section is missing AEP 3.x tab

Reply:
I was wondering how we wanted to handle this section for AEP 3.x since we have more public classes / services now. Should i just add documentation here for `AEPError`?

Reply:
Yes, that would be useful. The other services will be handled in other pages/sections.